### PR TITLE
Implement helper for create presentation

### PR DESCRIPTION
**Pull request by vineetpant**
## Description
This PR implements `helper_create_presentation` helper function

## Details
- Add `helper_create_presentation`
-  Update c and wasm bindings for function 
- Update cli for create_presentation helper
- Add Tests 
- Update versions.md
- Update release ci for wasm build updload.

### CODE REVIEW

File: src/helpers/presentation.rs [hunk 1]
```diff
@@ -108,6 +116,131 @@ impl<'a> Presentation<'a> {
...
+                })
+                .transpose()?;
+            let reveal_attributes = self
+                .get_reveal_attributes_indices_map(schema_did, revealed_attributes_parsed)
+                .await?;
+            for sub_proof in proof_request.sub_proof_requests.iter_mut() {
+                if &sub_proof.schema == schema_did {
+                    sub_proof.revealed_attributes = reveal_attributes
+                        .get(schema_did)
+                        .ok_or("schema attribute indices missing")
```
Comment (wulfraem) [inline src/helpers/presentation.rs]: Is there a reason for creating the error this was or could this be combined into a single `.ok_or_else` call?

File: src/helpers/presentation.rs [hunk 2]
```diff
@@ -108,6 +116,131 @@ impl<'a> Presentation<'a> {
...
+        let revealed = credential.credential_subject.clone();
+        revealed_properties_schema_map.insert(schema_did.to_owned(), revealed);
+
+        // get prover did
+        let prover = credential
+            .credential_subject
+            .id
+            .clone()
+            .ok_or("prover did required")
+            .map_err(|err| PresentationError::VadeEvanError(err.to_string()))?;
```
Comment (wulfraem) [inline src/helpers/presentation.rs]: Is there a reason for creating the error this was or could this be combined into a single `.ok_or_else` call?

File: src/helpers/presentation.rs [hunk 3]
```diff
@@ -108,6 +116,131 @@ impl<'a> Presentation<'a> {
...
+            PresentationError::to_deserialization_error("proof request", proof_request_str),
+        )?;
+        let matched_schema = proof_request
+            .sub_proof_requests
+            .clone()
+            .into_iter()
+            .filter(|sub_proof| &sub_proof.schema == schema_did)
+            .collect::<Vec<BbsSubProofRequest>>();
+
+        if matched_schema.len() <= 0 {
```
Comment (Tschuck) [inline src/helpers/presentation.rs]:
```suggestion
        if matched_schema.is_empty() {
```

File: src/helpers/presentation.rs [hunk 4]
```diff
@@ -108,6 +116,131 @@ impl<'a> Presentation<'a> {
...
+    ///
+    /// * `proof_request` - proof request for presentation
+    /// * `credential` - credential to be shared in presentation
+    /// * `master_secret` - user's master secret
+    /// * `signing_key` - users secp256k1 private signing key
+    /// * `revealed_attributes` - list of names of revealed attributes in specified schema,
+    ///
+    /// # Returns
+    /// * `Option<String>` - A `Presentation` as JSON
+    pub async fn create_presentation(
```
Comment (wulfraem) [inline src/helpers/presentation.rs:132]: Can you check the usage of `PresentationError::VadeEvanError`?

This error type is only intended to be used when a call against an internal `Vade` instance failed. For the other errors handled here, we could add a separate error for invalid input, etc. ^^

**Review by wulfraem — CHANGES_REQUESTED** [review]

File: src/helpers/presentation.rs (hunk 4 above)
Comment (vineetpant) [reply src/helpers/presentation.rs:132]: done, updated the code to use VadeEvanError only for vade calls.

File: src/helpers/presentation.rs (hunk 2 above)
Comment (vineetpant) [reply src/helpers/presentation.rs]: changed to ok_or_else()

File: src/helpers/presentation.rs (hunk 1 above)
Comment (vineetpant) [reply src/helpers/presentation.rs]: changed to ok_or_else()

**Review by wulfraem — APPROVED** [review]

**Review by Tschuck — APPROVED** [review]

**Pull request merged**